includes/blocks/class-sensei-course-outline-block.php

```diff
@@ -238,4 +236,82 @@ public function render_course_outline_block( $attributes ) {
 
 	}
 
+	/**
+	 * Initialize notices.
+	 *
+	 * @internal
+	 *
+	 * @since $$next-version$$
+	 */
+	public function frontend_notices() {
+		$post = get_post();
+
+		if ( ! is_object( $post ) || 'course' !== $post->post_type || is_admin() ) {
+			return;
+		}
+
+		$course_id            = $post->ID;
+		$structure            = Sensei_Course_Structure::instance( $course_id )->get( 'view' );
+		$has_draft            = $this->has_draft( $structure );
+		$edit_lessons_message = '';
+
+		// Complete message with CTA if user can edit the course.
+		if ( Sensei_Course::can_current_user_edit_course( $course_id ) ) {
+			$cta = sprintf(
+				'<a href="%s">%s</a>',
+				esc_url( get_edit_post_link( $course_id ) ?? '' ),
+				__( 'your lessons', 'sensei-lms' )
+			);
+
+			$edit_lessons_message = sprintf(
+				/* translators: %s: Edit course link. */
+				__( "When you're ready, let's publish %s in order to make them available to your students.", 'sensei-lms' ),
+				$cta
+			);
+		}
+
+		// Notice for empty structure. Notice that draft lessons don't return for students, so it's considered empty.
+		if ( empty( $structure ) ) {
+			$message = __( 'There are no published lessons in this course yet.', 'sensei-lms' );
+			if ( $edit_lessons_message ) {
+				$message = $edit_lessons_message . ' ' . $message;
+			}
+
+			Sensei()->notices->add_notice( $message, 'info', 'sensei-course-outline-no-content' );
+		}
+
+		// phpcs:ignore WordPress.Security.NonceVerification
+		if ( isset( $_GET['draftcourse'] ) && 'true' === $_GET['draftcourse'] ) {
+			// Notice when trying to register in a draft course.
+			$message = __( 'Cannot register for an unpublished course.', 'sensei-lms' );
+
+			if ( Sensei_Course::can_current_user_edit_course( $course_id ) ) {
+				$cta = sprintf(
+					'<a href="%s">%s</a>',
+					esc_url( get_edit_post_link( $course_id ) ?? '' ),
+					__( 'publish the course', 'sensei-lms' )
+				);
+
+				$publish_message = sprintf(
+					/* translators: %s: Link to publish the course. */
+					__( 'Please %s first.', 'sensei-lms' ),
+					$cta
+				);
+
+				$message .= ' ' . $publish_message;
+			}
+
+			Sensei()->notices->add_notice( $message, 'info', 'sensei-course-outline-drafts' );
+
+		} elseif ( $has_draft ) {
+			// Notice for draft lessons. It only happens for who can read private posts, otherwise draft lessons aren't returned.
+			$message = __( 'Unpublished lessons are only visible in preview mode.', 'sensei-lms' );
```

Comment:
```suggestion
			$message = __( 'Draft lessons are only visible in preview mode.', 'sensei-lms' );
```

I'd like to make this small tweak from the original requirements. 🙂